src/dbus/server/dbus_thread_object.cpp

```diff
@@ -556,5 +589,248 @@ otError DBusThreadObject::GetSupportedChannelMaskHandler(DBusMessageIter &aIter)
+    uint8_t              len = sizeof(data);
+    std::vector<uint8_t> networkData;
+
+    otNetDataGet(threadHelper->GetInstance(), /*stable=*/true, data, &len);
+    networkData = std::vector<uint8_t>(&data[0], &data[len]);
+    VerifyOrExit(DBusMessageEncodeToVariant(&aIter, networkData) == OTBR_ERROR_NONE, err = OT_ERROR_INVALID_ARGS);
+exit:
+    return err;
+}
+
+otError DBusThreadObject::GetLocalLeaderWeightHandler(DBusMessageIter &aIter)
+{
+    auto    threadHelper = mNcp->GetThreadHelper();
+    otError err          = OT_ERROR_NONE;
+    uint8_t weight       = otThreadGetLocalLeaderWeight(threadHelper->GetInstance());
+
+    VerifyOrExit(DBusMessageEncodeToVariant(&aIter, weight) == OTBR_ERROR_NONE, err = OT_ERROR_INVALID_ARGS);
+exit:
+    return err;
+}
+
+otError DBusThreadObject::GetChannelMonitorSampleCountHandler(DBusMessageIter &aIter)
+{
+#if OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE
+    auto     threadHelper = mNcp->GetThreadHelper();
+    otError  err          = OT_ERROR_NONE;
+    uint32_t cnt          = otChannelMonitorGetSampleCount(threadHelper->GetInstance());
+
+    VerifyOrExit(DBusMessageEncodeToVariant(&aIter, cnt) == OTBR_ERROR_NONE, err = OT_ERROR_INVALID_ARGS);
+exit:
+    return err;
+#else  // OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE
+    return OT_ERROR_NOT_IMPLEMENTED;
+#endif // OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE
+}
+
+otError DBusThreadObject::GetChannelMonitorChannelQualityMapHandler(DBusMessageIter &aIter)
+{
+#if OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE
+    auto                            threadHelper = mNcp->GetThreadHelper();
+    otError                         err          = OT_ERROR_NONE;
+    uint32_t                        channelMask  = otLinkGetSupportedChannelMask(threadHelper->GetInstance());
+    constexpr uint8_t               kNumChannels = sizeof(channelMask) * 8; // 8 bit per byte
+    std::vector<otbrChannelQuality> quality;
+
+    for (uint8_t i = 0; i < kNumChannels; i++)
+    {
+        if (channelMask & (1 << i))
+        {
+            uint16_t occupancy = otChannelMonitorGetChannelOccupancy(threadHelper->GetInstance(), i);
+
+            quality.emplace_back(otbrChannelQuality{i, occupancy});
+        }
+    }
+
+    VerifyOrExit(DBusMessageEncodeToVariant(&aIter, quality) == OTBR_ERROR_NONE, err = OT_ERROR_INVALID_ARGS);
+exit:
+    return err;
+#else  // OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE
+    return OT_ERROR_NOT_IMPLEMENTED;
+#endif // OPENTHREAD_CONFIG_CHANNEL_MONITOR_ENABLE
+}
+
+otError DBusThreadObject::GetChildTableHandler(DBusMessageIter &aIter)
+{
+    auto                       threadHelper = mNcp->GetThreadHelper();
+    otError                    err          = OT_ERROR_NONE;
+    uint16_t                   childIndex   = 0;
+    otChildInfo                childInfo;
+    std::vector<otbrChildInfo> childTable;
+
+    while (otThreadGetChildInfoByIndex(threadHelper->GetInstance(), childIndex, &childInfo) == OT_ERROR_NONE)
```

Comment:
Is ignoring the error intentional?

Reply:
Yes, the only error case is end of iteration.